crates/rpc-types/src/eth/transaction/receipt.rs

```diff
@@ -69,4 +71,85 @@ impl TransactionReceipt {
         }
         Some(self.from.create(nonce))
     }
+
+    /// Returns `true` if the transaction was successful.
+    /// A transaction is considered successful if the status code is `1`.
+    fn success(&self) -> bool {
+        match &self.status_code {
+            Some(status) => status.to::<u64>() == 1,
+            None => false,
+        }
+    }
+
+    /// Returns an iterator over the logs for prmitives type conversion.
+    fn logs_iter(
+        &self,
+    ) -> impl Iterator<Item = alloy_primitives::Log<alloy_primitives::LogData>> + '_ {
+        self.logs.iter().map(|log| {
+            alloy_primitives::Log::new_unchecked(log.address, log.topics.clone(), log.data.clone())
+        })
+    }
+}
+
+impl TryFrom<TransactionReceipt> for ReceiptWithBloom {
+    type Error = ConversionError;
+
+    fn try_from(tx_receipt: TransactionReceipt) -> Result<Self, Self::Error> {
+        let receipt_with_bloom = ReceiptWithBloom {
+            receipt: Receipt {
+                success: tx_receipt.success(),
+                cumulative_gas_used: tx_receipt.cumulative_gas_used.to::<u64>(),
+                logs: tx_receipt.logs_iter().collect_vec(),
+            },
+            bloom: tx_receipt.logs_bloom,
+        };
+        Ok(receipt_with_bloom)
+    }
+}
+
+impl TryFrom<TransactionReceipt> for ReceiptEnvelope {
```

Comment:
this can also be implemented for &TransactionReceipt, and I'd like to have them as functions of the `TransactionReceipt` typ

```
fn into_(self
fn as_(&self
```